### PR TITLE
Force to specify not empty secret for metrics endpoint

### PR DESCRIPTION
**Pull request by legionus**
@mfojtik looks good ?

### CODE REVIEW

File: pkg/cmd/dockerregistry/dockerregistry.go [hunk 1]
```diff
@@ -125,6 +125,9 @@ func Execute(configFile io.Reader) {
 
 	// Registry extensions endpoint provides prometheus metrics.
 	if extraConfig.Metrics.Enabled {
+		if len(extraConfig.Metrics.Secret) == 0 {
+			context.GetLogger(app).Fatalf("secret token for metrics endpoint cannot be empty")
```
Comment (mfojtik) [inline pkg/cmd/dockerregistry/dockerregistry.go]: maybe specify the path like: `the openshift.metrics.secret field cannot be empty when metrics are enabled`

**Comment by legionus** [comment]
@mfojtik  done

**Comment by mfojtik** [comment]
LGTM [merge]

@smarterclayton FYI

**Comment by mfojtik** [comment]
@legionus can you please rebase and then tag this again for merge? (1.6 rebase landed...)

**Comment by legionus** [comment]
[merge]

**Comment by legionus** [comment]
[merge]

**Comment by mfojtik** [comment]
[merge]

**Comment by legionus** [comment]
[merge] #14122

**Comment by smarterclayton** [comment]
Can you change `oadm registry` to set `prometheus.io/username=METRICS_USER` and `prometheus.io/password=METRICS_PASSWORD` and `prometheus.io/scrape=true` on the service at creation time?  I will be doing that for the router so that ootb the annotations will get set.

**Pull request merged**